Don't overload core with mempool tx requests

mononaut opened the pull request:
Fixes #4178 by fetching mempool transactions from Core sequentially instead of making 10k requests in parallel.

nymkappa commented on backend/src/api/transaction-utils.ts:
```diff
@@ -74,8 +74,18 @@ class TransactionUtils {
+            results.push(result);
+          }
+        } catch {
+          // skip failures
```
Would be good to log something in DEBUG, or do we already log in another function?

nymkappa replied:
Also for the dev who will land on this comment in a few months, could be good to elaborate a bit why we're fine skipping failures

mononaut replied:
we already log progress in the outer function (although those logs might be a bit sparse with a slow core backend)

junderw commented:
Perhaps using https://www.npmjs.com/package/p-limit might be better?

Basically it creates N slots for promises to run concurrently, then when one resolves, it fills the slot with the next promise in the queue.

(You pass a closure that returns the promise `limit(() => myPromise())` so that the promise doesn't start (send a request) until the queue puts it in a slot)

It's pretty useful.

junderw commented:
https://github.com/mempool/mempool/commit/ebf43bd074be315f524f32ccf88b742bc822f146

Joining together the two packages, It's only 142 lines of code. So I just added it with TypeScript definitions. Should be useful.

Feel free to merge it into your branch if you want.

32 concurrent seems like a reasonable amount for bitcoind... maybe a little lower might be good?

mononaut commented:
added, thanks!

32 concurrent requests still saturates the default Core maximum RPC work depth, so I've dropped that down to a more conservative 8, which should be fast enough while leaving plenty of spare room for other tasks and applications.

nymkappa reviewed (approved):
Tested ACK @ b9838fda8db62aae69a261ec8f15a5d5b10a7133
